packages/cloud-cognitive/src/components/EditTearsheet/EditTearsheet.js

```diff
@@ -0,0 +1,345 @@
+  influencerWidth: 'narrow',
+};
+
+export let EditTearsheet = forwardRef(
+  (
+    {
+      // The component props, in alphabetical order (for consistency).
+
+      backButtonText,
+      cancelButtonText,
+      children,
+      className,
+      description,
+      influencerWidth = defaults.influencerWidth,
+      initialStep,
+      label,
+      nextButtonText,
+      onClose,
+      onRequestSubmit,
+      open,
+      submitButtonText,
+      title,
+      verticalPosition = defaults.verticalPosition,
+      onHandleModalClick,
+
+      // Collect any other property values passed in.
+      ...rest
+    },
+    ref
+  ) => {
+    const [editTearsheetActions, setEditTearsheetActions] = useState([]);
+    const [shouldViewAll, setShouldViewAll] = useState(false);
+    const [currentStep, setCurrentStep] = useState(0);
+    const [isSubmitting, setIsSubmitting] = useState(false);
+    const [isDisabled, setIsDisabled] = useState(false);
+    const [onNext, setOnNext] = useState();
+    const [onMount, setOnMount] = useState();
+    const [stepData, setStepData] = useState([]);
+    const [firstIncludedStep, setFirstIncludedStep] = useState(1);
+    const [lastIncludedStep, setLastIncludedStep] = useState(null);
+
+    const previousState = usePreviousValue({ currentStep, open });
+    const contentRef = useRef();
+
+    useEffect(() => {
+      const firstItem =
+        stepData.findIndex((item) => item?.shouldIncludeStep) + 1;
+      const lastItem = lastIndexInArray(stepData, 'shouldIncludeStep', true);
+      if (firstItem !== firstIncludedStep) {
+        setFirstIncludedStep(firstItem);
+      }
+      if (lastItem !== lastIncludedStep) {
+        setLastIncludedStep(lastItem);
+      }
+    }, [stepData, firstIncludedStep, lastIncludedStep]);
+
+    useCreateComponentFocus({
+      previousState,
+      currentStep,
+      blockClass,
+      onMount,
+    });
+    useValidCreateStepCount(stepData.length, componentName);
+    useResetCreateComponent({
+      firstIncludedStep,
+      previousState,
+      setCurrentStep,
+      initialStep,
+      totalSteps: stepData?.length,
+      componentName,
+    });
+    useCreateComponentStepChange({
+      firstIncludedStep,
+      lastIncludedStep,
+      stepData,
+      onNext,
+      isSubmitDisabled: isDisabled,
+      setCurrentStep,
+      setIsSubmitting,
+      setShouldViewAll,
+      onClose,
+      onRequestSubmit,
+      componentName,
+      currentStep,
+      shouldViewAll,
+      backButtonText,
+      cancelButtonText,
+      submitButtonText,
+      nextButtonText,
+      isSubmitting,
+      componentBlockClass: blockClass,
+      setCreateComponentActions: setEditTearsheetActions,
+    });
+
+    const sideNavItems = [
+      { label: 'Topic Name' },
+      { label: 'Location' },
+      { label: 'Partitions' },
+      { label: 'Message retention' },
+    ];
+
+    const influencer = (
+      <div className="tearsheet-stories__dummy-influencer-block">
```

Comment:
i'm assuming this is storybook related? why is there a storybook specific class in the component?